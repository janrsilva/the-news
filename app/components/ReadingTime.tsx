import React from "react";

const ReadingTime: React.FC<{ size: number }> = ({ size }) => {
  const readingSpeedAvg = 800;
  const minutes = Math.ceil(size / readingSpeedAvg);

  return <small>{minutes} minutes read</small>;
};

export default ReadingTime;
