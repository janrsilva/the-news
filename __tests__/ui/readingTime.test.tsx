import { render } from "@testing-library/react";
import ReadingTime from "@/components/ReadingTime";

it("the reading time is rendered as 1 minute", () => {
  const { container } = render(<ReadingTime size={800} />);
  expect(container).toMatchSnapshot();
});

it("the reading time is rendered as 1 minute", () => {
  const { container } = render(<ReadingTime size={700} />);
  expect(container).toMatchSnapshot();
});

it("the reading time is rendered as 2 minutes", () => {
  const { container } = render(<ReadingTime size={850} />);
  expect(container).toMatchSnapshot();
});
