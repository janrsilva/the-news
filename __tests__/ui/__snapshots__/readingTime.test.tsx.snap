// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the reading time is rendered as 1 minute 1`] = `
<div>
  <small>
    1
     minutes read
  </small>
</div>
`;

exports[`the reading time is rendered as 1 minute 2`] = `
<div>
  <small>
    1
     minutes read
  </small>
</div>
`;

exports[`the reading time is rendered as 2 minutes 1`] = `
<div>
  <small>
    2
     minutes read
  </small>
</div>
`;
